gtsfm/scene_optimizer.py

```diff
@@ -122,6 +122,7 @@ def create_computation_graph(
             descriptors_graph_list += [delayed_descs]
 
         # Estimate two-view geometry and get indices of verified correspondences.
+        matcher_reports_dict = {}
```

Comment:
Why are we creating and passing around an additional report that just holds the number of matches and inlier ratio, both of which are already stored in the `TwoViewEstimationReport`.

Reply:
The main idea behind this is the following direction from @dellaert :
(1) Each blue box in the graph should have its own table.
(2) We need to be totally functional in our programming. (We cannot modify objects)
(3) Each table can only be returned by one blue box.
(4) No two blue boxes can create a single gray box.
(5) The graph must determine the code/data types exactly.

<img width="1480" alt="Screen Shot 2021-11-05 at 8 56 37 AM" src="https://user-images.githubusercontent.com/16724970/140513851-81e85c98-048c-4303-b7a6-6a1cc9bb08f5.png">

Reply:
As we talked about offline, I think all of this can be achieved without creating a redundant data type for each blue box. If we continue on this trajectory, we will be passing around 7-8 different reports, which will be extremely cumbersome. 

I think we should continue brainstorming.